test/unit/spec/PasswordComplexityUtil_spec.js

```diff
@@ -0,0 +1,293 @@
+          expect(this.isEnabled(0)).toBeFalsy();
+        });
+      });
+
+      describe('minLowerCase', function () {
+        beforeEach(function () {
+          this.isEnabled = this.complexities['minLowerCase'].isEnabled;
+        });
+
+        it('1 is enabled', function () {
+          expect(this.isEnabled(1)).toBeTruthy();
+        });
+
+        it('0 is disabled', function () {
+          expect(this.isEnabled(0)).toBeFalsy();
+        });
+      });
+
+      describe('minUpperCase', function () {
+        beforeEach(function () {
+          this.isEnabled = this.complexities['minUpperCase'].isEnabled;
+        });
+
+        it('1 is enabled', function () {
+          expect(this.isEnabled(1)).toBeTruthy();
+        });
+
+        it('0 is disabled', function () {
+          expect(this.isEnabled(0)).toBeFalsy();
+        });
+      });
+
+      describe('minNumber', function () {
+        beforeEach(function () {
+          this.isEnabled = this.complexities['minNumber'].isEnabled;
+        });
+
+        it('1 is enabled', function () {
+          expect(this.isEnabled(1)).toBeTruthy();
+        });
+
+        it('0 is disabled', function () {
+          expect(this.isEnabled(0)).toBeFalsy();
+        });
+      });
+
+      describe('minSymbol', function () {
+        beforeEach(function () {
+          this.isEnabled = this.complexities['minSymbol'].isEnabled;
+        });
+
+        it('1 is enabled', function () {
+          expect(this.isEnabled(1)).toBeTruthy();
+        });
+
+        it('0 is disabled', function () {
+          expect(this.isEnabled(0)).toBeFalsy();
+        });
+      });
+
+      describe('excludeUsername', function () {
+        beforeEach(function () {
+          this.isEnabled = this.complexities['excludeUsername'].isEnabled;
+        });
+
+        it('true is enabled', function () {
+          expect(this.isEnabled(true)).toBeTruthy();
+        });
+
+        it('false is disabled', function () {
+          expect(this.isEnabled(false)).toBeFalsy();
+        });
+      });
+    });
+
+    describe('doesComplexityMeet', function () {
+      beforeEach(function () {
+        this.complexities = PasswordComplexityUtil.complexities;
+      });
+
+      describe('minLength is 10 characters', function () {
+        beforeEach(function () {
+          this.doesComplexityMeet = _.partial(this.complexities['minLength'].doesComplexityMeet, 10);
+        });
+
+        it('10 characters password meets the complexity', function () {
+          expect(this.doesComplexityMeet('1234567890')).toBeTruthy();
+        });
+
+        it('9 characters password does not meet the complexity', function () {
+          expect(this.doesComplexityMeet('123456789')).toBeFalsy();
+        });
+
+        it('empty password does not meet the complexity', function () {
+          expect(this.doesComplexityMeet('')).toBeFalsy();
+        });
+      });
+
+      describe('minLowerCase is 2 characters', function () {
```

Comment:
can we have all the minLowerCase tests in the same describe block for minLowerCase?
Also, this should not be ```is 2 characters``` but more a describe block inside `minLowerCase` that reads ```complexity validation``` or so